components/script/script_module.rs

```diff
@@ -731,7 +733,7 @@ impl ModuleTree {
                         url.clone(),
                         visited_urls.clone(),
                         destination.clone(),
-                        Referrer::Client,
+                        referrer.clone(),
```

Comment:
We can remove this argument from fetch_single_module_script and use global.get_referrer() in that code directly instead.